src/ocm/hooks/useClusterSupportedPlatforms.ts

```diff
@@ -17,12 +25,10 @@ export default function useClusterSupportedPlatforms(clusterId: string) {
   });
 
   const isLoading = !error && !data;
-  // Platform integration is supported
-  // if there is another platform type
-  // besides 'baremetal' or 'none', in the returned data.
-  const isPlatformIntegrationSupported =
-    !isLoading &&
-    (data?.filter((platform) => !['none', 'baremetal'].includes(platform)) || [])?.length > 0;
+  const supportedPlatformIntegration =
```

Comment:
Based on what @jkilzi  said, this algorithm is wrong because it won't catch cases where the values are mixed. It will think platform integration is enabled when it shouldn't.

So to allow integration only when all hosts originate from the same platform, the check must be exclusive.

Maybe you could use `groupBy` from lodash to group the values and see if there is more than 1 (didn't check, there may be better ways to do it).

Reply:
The platform integration is supported when all the entries in the payload received from /v2/supported-platforms are from the same provider (either vSphere or Nutanix). Perhaps all that's needed is to remove duplicated entries from the response (creating a Set from the array received would help) and then check if there is a single entry. If that entry is either `nutanix`or `vsphere` then the platform is supported and you'll be able to tell also the platform provider.